### PR TITLE
show staff instead of user badge when reported by staff

### PR DESCRIPTION
**Pull request by blackcathacker**
## What does this PR do?

Fixes issue #1013 . Displays 'Staff' in badge when Admin, Moderator, or Staff report a comment instead of just 'User' I know this issue was claimed by someone else, but...enjoying this project and looking for my hacktoberfest numbers. :) Feel free to close if I'm overreaching here.

## How do I test this PR?

report a comment as a staff member and see the comment in the reported queue says 'Staff' in the badge instead of 'User'.

image

### CODE REVIEW

**Review by kgardnr — DISMISSED** [review]

File: client/coral-admin/src/components/CommentLabels.js [hunk 1]
```diff
@@ -6,10 +6,19 @@ import FlagLabel from 'coral-ui/components/FlagLabel';
...
+
 function isUserFlagged(actions) {
   return actions.some((action) => action.__typename === 'FlagAction' && action.user);
 }
 
+function getUserFlaggedType(actions) {
+  return actions
+    .filter((action) => action.__typename === 'FlagAction' && action.user)
+    .map((action) => action.user.roles.some((role) => staffRoles.includes(role)))
+    .some((staff) => staff) ? 'Staff' : 'User';
```
Comment (alinhle) [inline client/coral-admin/src/components/CommentLabels.js]: This chain of methods seems a bit complicated. Why not
```js
function getUserFlaggedType(actions) {
  return actions
    .some((action) =>
      action.__typename === 'FlagAction' &&
      action.user &&
      action.user.roles.some((role) => staffRoles.includes(role))
    ) ? 'Staff' : 'User';
}
```

**Review by alinhle — CHANGES_REQUESTED** [review]
Looking good feature-wise. 
See my small remark below.

**Comment by kgardnr** [comment]
Hey @blackcathacker just a small comment above and then we can merge it - thank you! 👍

File: client/coral-admin/src/components/CommentLabels.js (hunk 1 above)
Comment (blackcathacker) [reply client/coral-admin/src/components/CommentLabels.js]: That is cleaner. Updated.

**Review by alinhle — APPROVED** [review]
Awesome thanks @blackcathacker

**Pull request merged**